backend/extensionApi/extensionApi.go

```diff
@@ -0,0 +1,146 @@
+package extensionApi
+
+import (
+	"fmt"
+	"github.com/robertkrimen/otto"
+	"os"
+)
+
+func GetExtensionFromFile(fileName string, sqlClient SimpleSqlClient) (Extension, error) {
+	vm := otto.New()
+	sqlClientJs, err := addSqlClient(vm, sqlClient)
+	if err != nil {
+		return nil, err
+	}
+	extensionJs, err := loadExtension(vm, fileName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load extension %v. Cause: %v", fileName, err.Error())
+	}
+	extensionName, err := readRequiredStringProperty(extensionJs, "name")
+	if err != nil {
+		return nil, err
+	}
+	extension := ExtensionImpl{*extensionName, extensionJs, sqlClientJs}
+	return extension, nil
+}
+
+func readRequiredStringProperty(extensionJs *otto.Object, propertyName string) (*string, error) {
+	result, err := extensionJs.Get(propertyName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read required extension property %v. Cause: %v", propertyName, err.Error())
+	}
+	stringResult, err := result.ToString()
+	if err != nil {
+		return nil, fmt.Errorf("invalid value for extension property %v. The value must be a string. Cause: %v", propertyName, err.Error())
+	}
+	return &stringResult, nil
+}
+
+func loadExtension(vm *otto.Otto, fileName string) (*otto.Object, error) {
+	const extensionVariable = "installedExtension"
+	err := vm.Set(extensionVariable, otto.NullValue())
+	if err != nil {
```

Comment:
oh, you actually face the same issue, we face in the UDFs, you need to check every call to the VM. Not sure, if this is any issue here, we gave each of them a unique error code, such that we can identify where it fails.

Reply:
created an issue for unified error-codes: https://github.com/exasol/extension-manager/issues/8